test/distributed/test_c10d_nccl.py

```diff
@@ -420,6 +420,47 @@ def allgather_base(output_t, input_t):
             # fails the check because the dtype is different
             allgather_base(output_t, tensor)
 
+    @requires_nccl()
+    def test_gather_ops(self):
```

Comment:
I think we need additional unit tests here similar to what we have in https://github.com/pytorch/pytorch/blob/master/test/distributed/test_c10d_gloo.py. Look at the following tests in that file: [test_gather_checks](https://github.com/pytorch/pytorch/blob/master/test/distributed/test_c10d_gloo.py#L809), [test_gather_basics](https://github.com/pytorch/pytorch/blob/master/test/distributed/test_c10d_gloo.py#L914) and [test_gather_stress](https://github.com/pytorch/pytorch/blob/master/test/distributed/test_c10d_gloo.py#L957). We should have similar tests in NCCL.

Reply:
This test is under `ProcessGroupNCCLTest` which had only a single process. We should also have tests under `class CommTest` which tests multiple processes using NCCL.

Reply:
Need to address my last comment here to ensure we have tests that run with multiple processes.